oauth-client/src/main/java/io/streamnative/pulsar/handlers/kop/security/oauth/OauthLoginCallbackHandler.java

```diff
@@ -68,6 +74,8 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
                 } catch (KafkaException e) {
                     throw new IOException(e.getMessage(), e);
                 }
+            } else if (callback instanceof SaslExtensionsCallback) {
```

Comment:
In which case could `callback` be `SaslExtensionsCallback`? It seems to leverage the [KIP-342](https://cwiki.apache.org/confluence/display/KAFKA/KIP-342%3A+Add+support+for+Custom+SASL+extensions+in+OAuthBearer+authentication) feature, which is only available for Kafka client 2.1 or later.

If so, we need to document that the latest `oauth-client` module could only be used for Kafka client 2.1 or later.

Reply:
Yes, we need to use Kafka client 2.1 or later.